pkg/clients/dclient/discovery.go

```diff
@@ -152,7 +152,13 @@ func (c serverPreferredResources) findResource(apiVersion string, kind string) (
 					logger.Error(err, "failed to parse GV", "groupVersion", serverResource.GroupVersion)
 					return nil, schema.GroupVersionResource{}, err
 				}
-
+				// We potentially need to fix Group and Version with what the list is for
```

Comment:
Maybe we should have this change in a separate PR ?

Reply:
What does this mean?

Reply:
Is the group/version for the type list the same as the group/version of the item in the list?

I'm listing out clusterroles below and the list G/V is `v1` while the ClusterRole belongs to `rbac.authorization.k8s.io/v1`:
```
# kubectl get clusterrole -o yaml | more
apiVersion: v1
items:
- aggregationRule:
    clusterRoleSelectors:
    - matchLabels:
        rbac.authorization.k8s.io/aggregate-to-admin: "true"
  apiVersion: rbac.authorization.k8s.io/v1
  kind: ClusterRole
  metadata:
    annotations:
...
```

Reply:
We are using discovery here, not listing resources.

Reply:
See below:
```
// APIResourceList is a list of APIResource, it is used to expose the name of the
// resources supported in a specific group and version, and if the resource
// is namespaced.
type APIResourceList struct {
	TypeMeta `json:",inline"`
	// groupVersion is the group and version this APIResourceList is for.
	GroupVersion string `json:"groupVersion" protobuf:"bytes,1,opt,name=groupVersion"`
	// resources contains the name of the resources and if they are namespaced.
	APIResources []APIResource `json:"resources" protobuf:"bytes,2,rep,name=resources"`
}
```

Then
```
// APIResource specifies the name of a resource and whether it is namespaced.
type APIResource struct {
	// name is the plural name of the resource.
	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
	// singularName is the singular name of the resource.  This allows clients to handle plural and singular opaquely.
	// The singularName is more correct for reporting status on a single item and both singular and plural are allowed
	// from the kubectl CLI interface.
	SingularName string `json:"singularName" protobuf:"bytes,6,opt,name=singularName"`
	// namespaced indicates if a resource is namespaced or not.
	Namespaced bool `json:"namespaced" protobuf:"varint,2,opt,name=namespaced"`
	// group is the preferred group of the resource.  Empty implies the group of the containing resource list.
	// For subresources, this may have a different value, for example: Scale".
	Group string `json:"group,omitempty" protobuf:"bytes,8,opt,name=group"`
	// version is the preferred version of the resource.  Empty implies the version of the containing resource list
	// For subresources, this may have a different value, for example: v1 (while inside a v1beta1 version of the core resource's group)".
	Version string `json:"version,omitempty" protobuf:"bytes,9,opt,name=version"`
```

The important comment on `Group` and `Version` fields is `Empty implies the group of the containing resource list.`

Reply:
Ok thanks! I was just navigated to this comment.

cc @Vyom-yadav regarding this comment, it might impact how we identify subresources:
```
	// For subresources, this may have a different value, for example: Scale".
```

Reply:
@realshuting We already handle this at:
https://github.com/Vyom-Yadav/kyverno/blob/5bcc1d23619b7633c2c8401e649ea3d2e5092ff5/pkg/clients/dclient/discovery.go#L158-L161